charts/tfy-agent/values.yaml

```diff
@@ -1,123 +1,296 @@
+  # the key must be token
+  # if this is set it will override config.clusterToken
+  clusterTokenSecret: ""
+
+  # url of the control plane
+  # format: https://<>
+  controlPlaneURL: ""
+
+  # name of the tenant where your control plane is hosted
+  tenantName: ""
+
+  # nats port
+  natsPort: '443'
+
+  # configurations for opencost
+  opencost:
+    # time in milliseconds for opencost scraping
+    pollInterval: '180000'
+    # endpoint to connect to opencost
+    endpoint: http://opencost.opencost.svc.cluster.local:9090
+
+  # configurations for prometheus
+  prometheus:
+    # time in milliseconds for prometheus scraping config
+    pollInterval: '60000'
+    # endpoint to connect to prometheus
+    endpoint: http://prometheus-operated.prometheus.svc.cluster.local:9090
+  
+  alertURL: "https://auth.truefoundry.com"
+
+  nodeEnv: production
 
 imagePullSecrets: []
 nameOverride: ""
 fullnameOverride: ""
 
-serviceAccount:
-  # Specifies whether a service account should be created
-  create: true
-  # Annotations to add to the service account
+# configuration related to tfy-agent
+tfyAgent:
+
+  ## Deploy tfyAgent
+  #
+  enabled: true
+
+  ## Number of replicas of tfyAgent
+  #
+  replicaCount: 1
+
+  ## Annotations for Alertmanager
+  ##
   annotations: {}
-  # The name of the service account to use.
-  # If not set and create is true, a name is generated using the fullname template
-  name: "tfy-agent"
 
+  ## Service account for Alertmanager to use.
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/
+  ##
+  serviceAccount:
+    # Specifies whether a service account should be created
+    create: true
+    # Annotations to add to the service account
+    annotations: {}
+    # The name of the service account to use.
+    # If not set and create is true, a name is generated using the fullname template
+    name: ""
+    # enabled automounting of service account token inside the pod
+    automountServiceAccountToken: true
+
+  ## Configuration for tfyAgent service
+  ##
+  service:
+    annotations: {}
+    labels: {}
+    clusterIP: ""
+
+    ## Port for tfyAgent Service to listen on
+    ##
+    port: 3000
+    ## Port to expose on each node
+    ## Only used if service.type is 'NodePort'
+    ##
+    nodePort: ""
+    ## List of IP addresses at which the tfyAgent server service is available
+    ## Ref: https://kubernetes.io/docs/user-guide/services/#external-ips
+    ##
+
+    ## If you want to make sure that connections from a particular client are passed to the same Pod each time
+    ## Accepts 'ClientIP' or ''
+    ##
+    sessionAffinity: ""
+
+    ## Service type
+    ##
+    type: ClusterIP
 
-envSecretName: tfy-agent-env-secret
-podAnnotations: {}
+  ## Image of tfyAgent
+  ##
+  image:
+    repository: truefoundrycloud/tfy-agent
+    pullPolicy: IfNotPresent
+    # Overrides the image tag whose default is the chart appVersion.
+    tag: "a9af8a51d8b0373f4d0c469119ed88a488c5709f"
```

Comment:
Is this going to be automatically updated via our existing CI system? Or will this be manual?

https://github.com/truefoundry/tfy-agent/blob/a9af8a51d8b0373f4d0c469119ed88a488c5709f/.github/workflows/build-n-publish.yaml#L61